projectional/src/main/java/jetbrains/jetpad/projectional/cell/mapping/CellProvider.java

```diff
@@ -77,8 +77,12 @@ public void remove() {
   }
 
   public List<Cell> getCellsOnPath(Iterator<?> sourcePath) {
-    Object actualSource = sourcePath.next();
-    return lookupCells(actualSource, actualSource, sourcePath);
+    if (sourcePath.hasNext()) {
+      Object actualSource = sourcePath.next();
+      return lookupCells(actualSource, actualSource, sourcePath);
+    } else {
+      return Collections.emptyList();
```

Comment:
This line (emptyList()) is not hit by CellProviderTest tests. Also the tests pass without this change.